common/inc/com/centreon/common/process.hh

```diff
@@ -0,0 +1,186 @@
+struct boost_process;
+}  // namespace detail
+
+/**
+ * @brief This class allow to exec a process asynchronously.
+ * It's a base class. If you want to get stdin and stdout returned data, you
+ * must inherit from this and override on_stdout_read and on_stderr_read
+ * You can call start_process at any moment, if a process is already running,
+ * it's killed
+ * As we can start a process at any moment, all handlers take a caller in
+ * parameter, if this caller is not equal to current _proc, we do nothing.
+ * When handler like on_stdout_read are called, _protect is already locked
+ */
+class process : public std::enable_shared_from_this<process> {
+  std::shared_ptr<asio::io_context> _io_context;
+  std::shared_ptr<spdlog::logger> _logger;
+
+  std::string _exe_path;
+  std::vector<std::string> _args;
+
+  std::deque<std::shared_ptr<std::string>> _stdin_write_queue;
+  bool _write_pending = false;
+
+  std::shared_ptr<detail::boost_process> _proc;
+
+  int _exit_status = 0;
+
+  std::mutex _protect;
+
+  void stdin_write_no_lock(const std::shared_ptr<std::string>& data);
+  void stdin_write(const std::shared_ptr<std::string>& data);
+
+  void stdout_read();
+  void stderr_read();
+
+ protected:
+  char _stdout_read_buffer[0x1000];
+  char _stderr_read_buffer[0x1000];
+
+  virtual void on_stdout_read(const boost::system::error_code& err,
+                              size_t nb_read);
+  virtual void on_stderr_read(const boost::system::error_code& err,
+                              size_t nb_read);
+
+  virtual void on_process_end(const boost::system::error_code& err,
+                              int raw_exit_status);
+
+  virtual void on_stdin_write(const boost::system::error_code& err);
+
+ public:
+  template <typename string_iterator>
+  process(const std::shared_ptr<boost::asio::io_context>& io_context,
+          const std::shared_ptr<spdlog::logger>& logger,
+          const std::string& exe_path,
+          string_iterator arg_begin,
+          string_iterator arg_end);
+
+  template <typename args_container>
+  process(const std::shared_ptr<boost::asio::io_context>& io_context,
+          const std::shared_ptr<spdlog::logger>& logger,
+          const std::string& exe_path,
+          const args_container& args);
+
+  template <typename string_type>
+  process(const std::shared_ptr<boost::asio::io_context>& io_context,
+          const std::shared_ptr<spdlog::logger>& logger,
+          const std::string& exe_path,
+          const std::initializer_list<string_type>& args);
+
+  process(const std::shared_ptr<boost::asio::io_context>& io_context,
+          const std::shared_ptr<spdlog::logger>& logger,
+          const std::string& cmd_line);
+
+  virtual ~process() = default;
+
+  template <typename string_class>
+  void write_to_stdin(const string_class& content);
+
+  void start_process();
+
+  void kill();
+
+  int get_exit_status() const { return _exit_status; }
+  const std::string get_exe_path() const { return _exe_path; }
```

Comment:
So you don't want to return a reference to the string... ???

Reply:
done